### DISCUSSION TITLE
Is it possible to redirect SDK internal logs to callback function or similar ?

### DISCUSSION THREAD

**Discussion by kiranpradeep** [aws/aws-iot-device-sdk-js-v2#482] [Q&A] [answered]
Currently, SDK internal logs are written to stderr. So we use `2>&1` like syntax to capture those logs. Is there a way to get those logs to a JS callback method or similar? A reference is [xlogging_set_log_function](https://github.com/Azure/azure-iot-sdk-c/issues/1032#issuecomment-509137584) call in azure-iot-sdk-c

[1] https://github.com/Azure/azure-iot-sdk-c/issues/380#issuecomment-366314638

**Comment by bretambrose** [comment, marked as answer]
The node logger writes log lines to Process._rawDebug:

https://github.com/awslabs/aws-crt-nodejs/blob/main/source/logger.c#L257-L280

That connection cannot be changed, but it may be possible to monkey-patch Process._rawDebug to something else or configure node to route things differently.
